### PR TITLE
Fix incorrect handling of BeginXFunctionDefinition in OperationMutator

### PR DESCRIPTION
**Pull request by amarekano**

### CODE REVIEW

File: Sources/Fuzzilli/Mutators/OperationMutator.swift [hunk 1]
```diff
@@ -169,8 +169,18 @@ public class OperationMutator: BaseInstructionMutator {
             } else {
                 newOp = BeginFor(comparator: op.comparator, op: chooseUniform(from: allBinaryOperators))
             }
-        case let op as BeginAnyFunctionDefinition:
-            newOp = BeginAnyFunctionDefinition(signature: op.signature, isStrict: !op.isStrict)
+        case let op as BeginPlainFunctionDefinition:
+                newOp = BeginPlainFunctionDefinition(signature: op.signature, isStrict: !op.isStrict)
```
Comment (saelo) [inline Sources/Fuzzilli/Mutators/OperationMutator.swift:173]: Nit: wrong indention here and the line below

**Review by saelo — COMMENTED** [review]
Ah, nice catch! So this would've left programs with `BeginAnyFunctionDefinition` operations in them, which then probably crashed the Lifter?

File: Sources/Fuzzilli/Mutators/OperationMutator.swift (hunk 1 above)
Comment (amarekano) [reply Sources/Fuzzilli/Mutators/OperationMutator.swift]: fixed

**Comment by amarekano** [comment]
> which then probably crashed the Lifter?
This this is the bit that confused me! I haven't noticed any crashes in the Lifter but I suspect the swift compiler matches BeginAnyFunction with BeginPlainFunction and that probably why we don't see any crashes. Or it may be that the mutation rarely triggered so this crash does not occur as often.

**Comment by saelo** [comment]
Ah, actually I think the problem might be that `BeginAnyFunctionDefinition` isn't marked with `.isParametric` and so will never be mutated by that Mutator? At least in my local tests, I never see it being mutated so far.

**Pull request merged**